griptape/drivers/vector/qdrant_vector_store_driver.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import annotations
+from typing import Optional, Any
+from collections.abc import Iterable, Sequence, Generator
+from attrs import define, field
+import uuid
+from itertools import islice
+from griptape.drivers import BaseVectorStoreDriver
+from griptape.artifacts import TextArtifact
+
+from qdrant_client.http import models as rest
+
+VECTOR_NAME = None
+BATCH_SIZE = 64
+DEFAULT_DISTANCE = "COSINE"
+
+
+@define
+class QdrantVectorStoreDriver(BaseVectorStoreDriver):
+    """A Vector Store Driver for Qdrant Vector DB."""
+
+    location: Optional[str] = field(default=None, kw_only=True, metadata={"serializable": True})
+    url: Optional[str] = field(default=None, kw_only=True, metadata={"serializable": True})
+    host: Optional[str] = field(default=None, kw_only=True, metadata={"serializable": True})
+    port: int = field(default=6333, kw_only=True, metadata={"serializable": True})
+    grpc_port: int = field(default=6334, kw_only=True, metadata={"serializable": True})
+    prefer_grpc: bool = field(default=False, kw_only=True, metadata={"serializable": True})
+    force_recreate: bool = field(default=True, kw_only=True, metadata={"serializable": True})
+    distance: str = field(default=DEFAULT_DISTANCE, kw_only=True, metadata={"serializable": True})
+    collection_name: str = field(kw_only=True, metadata={"serializable": True})
+    vector_name: Optional[str] = VECTOR_NAME
+    content_payload_key: str = field(default="data", kw_only=True, metadata={"serializable": True})
+
+    def __attrs_post_init__(self) -> None:
+        from qdrant_client import QdrantClient, AsyncQdrantClient
+
+        if self.location == ":memory:":
+            self.client = AsyncQdrantClient(
+                location=self.location,
+                url=self.url,
+                host=self.host,
+                port=self.port,
+                prefer_grpc=self.prefer_grpc,
+                grpc_port=self.grpc_port,
+            )
+        else:
+            self.client = QdrantClient(
+                location=self.location,
+                url=self.url,
+                host=self.host,
+                port=self.port,
+                prefer_grpc=self.prefer_grpc,
+                grpc_port=self.grpc_port,
+            )
+        # Ensure the collection exists and has the correct configuration
+        self._create_collection(self.embedding_driver)
+
+    def delete_vector(self, vector_id: str) -> None:
+        """
+        Delete vectors from the Qdrant collection based on their IDs.
+
+        Parameters:
+        - vector_ids: Optional list of vector IDs to delete.
+        """
```

Comment:
Docstring should be formatted according to [Google's style guide](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods).